filesystem/mountpoint.go

```diff
@@ -57,79 +49,179 @@ var (
 	uuidDirectory = "/dev/disk/by-uuid"
 )
 
-// getMountInfo populates the Mount mappings by parsing the filesystem
-// description file using the getmntent functions. Returns ErrBadLoad if the
-// Mount mappings cannot be populated.
-func getMountInfo() error {
-	if mountsInitialized {
-		return nil
+// Unescape octal-encoded escape sequences in a string from the mountinfo file.
+// The kernel encodes the ' ', '\t', '\n', and '\\' bytes this way.
+// This function exactly inverts what the kernel does, including by preserving
+// invalid UTF-8, unlike strconv.UnquoteChar().
+func unescapeString(str string) string {
+	escapedStr := make([]byte, len(str))
+	j := 0
+	for i := 0; i < len(str); i++ {
+		b := str[i]
+		if b == '\\' && i+3 < len(str) &&
+			str[i+1] >= '0' && str[i+1] < '8' &&
+			str[i+2] >= '0' && str[i+2] < '8' &&
+			str[i+3] >= '0' && str[i+3] < '8' {
+			b = ((str[i+1] - '0') << 6) |
+				((str[i+2] - '0') << 3) |
+				(str[i+3] - '0')
```

Comment:
I agree with not having `strconv` handle all of the escaping, but it probably should handle the octal encoding. Something like:
```go
func unescapeString(str string) string {
	escapedStr := make([]byte, len(str))
	j := 0
	for i := 0; i < len(str); i++ {
		b := str[i]
		if b == '\\' && i+4 <= len(str) {
			if parsed, err := strconv.ParseInt(str[i+1:i+4], 8, 8); err == nil {
				b = parsed
				i += 3
			}
		}
		escapedStr[j] = b
		j++
	}
	return string(escapedStr[:j])
}
```
or if we use `strings.Builder`
```go
func unescapeString(str string) string {
	var sb strings.Builder
	for i := 0; i < len(str); i++ {
		b := str[i]
		if b == '\\' && i+4 <= len(str) {
			if parsed, err := strconv.ParseInt(str[i+1:i+4], 8, 8); err == nil {
				b = parsed
				i += 3
			}
		}
		sb.WriteByte(b)
	}
	return sb.String()
}
```